src/likelihoods/negativebinomial.jl

```diff
@@ -0,0 +1,26 @@
+"""
+    NegBinomialLikelihood(r::Real=1.0, l=logistic)
+
+Negative binomial likelihood with number of successes `r`.
+
+```math
+    p(k|f) = \\operatorname{NB}(k | r, l(f))
+```
+On calling, this would return a negative binomial distribution with `r` successes and probability of success equal to `l(f)`.
+Note the distinction of the role of parameter `r` to the Wikipedia definition.
+"""
+struct NegBinomialLikelihood{T<:Real,Tl<:AbstractLink} <: AbstractLikelihood
```

Comment:
Maybe we could use the name `NegativeBinomialLikelihood`, to be consistent with the name of the distribution? I guess also it's more common to dispatch on the link, so
```suggestion
struct NegativeBinomialLikelihood{Tl<:AbstractLink,T<:Real} <: AbstractLikelihood
```
might be more useful.

Reply:
I think this makes sense.

Reply:
Do you suggest a type parameter order different from e.g. `GammaLikelihood` on purpose?

Reply:
Yes, since we are not happy with the order in `GammaLikelihood`: https://github.com/JuliaGaussianProcesses/GPLikelihoods.jl/pull/41